lib/dry/transaction/step_failure.rb

```diff
@@ -12,6 +12,10 @@ def method_missing(name, *args, &block)
         @__object.send(name, *args, &block)
       end
 
+      def respond_to_missing?(*args)
+        @__object.respond_to?(*args)
```

Comment:
@AMHOL @timriley I guess it is better to replace `send` with `public_send` because there is no point in bypassing method access check via `StepFailure` object.